nncf/torch/tensor_statistics/collectors.py

```diff
@@ -11,243 +11,182 @@
+
 
-    def get_statistics(self) -> Dict[ReductionShape, TensorStatistic]:
-        if self._collected_samples == 0:
-            raise StatisticsNotCollectedError()
-        return self._get_statistics()
+class PTNNCFCollectorTensorProcessor(NNCFCollectorTensorProcessor):
+    """
+    A realization of the processing methods set for PTNNCFTensors.
+    """
 
-    @abstractmethod
-    def _get_statistics(self) -> Dict[ReductionShape, TensorStatistic]:
-        pass
+    @staticmethod
+    def reduce_min(x: NNCFTensor, axis: Union[int, tuple]) -> NNCFTensor:
+        return PTNNCFTensor(torch.amin(x.tensor, dim=axis))
 
-    def enable(self):
-        self._enabled = True
+    @staticmethod
+    def reduce_max(x: NNCFTensor, axis: Union[int, tuple]) -> NNCFTensor:
+        return PTNNCFTensor(torch.amax(x.tensor, dim=axis))
 
-    def disable(self):
-        self._enabled = False
+    @staticmethod
+    def abs(x: NNCFTensor) -> NNCFTensor:
+        return PTNNCFTensor(torch.abs(x.tensor))
 
-    def reset(self):
-        self._collected_samples = 0
-        self._reset()
+    @staticmethod
+    def min(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
+        return PTNNCFTensor(torch.min(x1.tensor, x2.tensor))
 
-    @abstractmethod
-    def _reset(self):
-        pass
+    @staticmethod
+    def max(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
+        return PTNNCFTensor(torch.max(x1.tensor, x2.tensor))
 
-    def collected_samples(self) -> int:
-        return self._collected_samples
+    @staticmethod
+    def mean(x: NNCFTensor, axis: Union[int, tuple]) -> NNCFTensor:
+        return PTNNCFTensor(x.tensor.mean(dim=axis))
 
+    @staticmethod
+    def stack(x: Union[List[NNCFTensor], Deque[NNCFTensor]], axis: int = 0) -> NNCFTensor:
+        x = [t.tensor for t in x]
+        return PTNNCFTensor(torch.stack(x, dim=axis))
 
-class StatisticsNotCollectedError(Exception):
-    pass
+    @staticmethod
+    def unstack(x: NNCFTensor, axis: int = 0) -> List[NNCFTensor]:
+        tensor_list = torch.unbind(x.tensor, dim=axis)
+        return [PTNNCFTensor(t) for t in tensor_list]
 
 
-class OnlineTensorStatisticCollector(TensorStatisticCollectorBase, ABC):
-    pass
+class PTMinMaxStatisticCollector(MinMaxStatisticCollector):
+    def __init__(self, use_abs_max: bool, reduction_shape: ReductionShape, output_shape: ReductionShape,
+                 num_samples: int = None):
+        super().__init__(use_abs_max, reduction_shape, num_samples)
+        self._output_shape = output_shape
 
+    @staticmethod
+    def _get_processor() -> NNCFCollectorTensorProcessor:
+        return PTNNCFCollectorTensorProcessor()
 
-class OfflineTensorStatisticCollector(TensorStatisticCollectorBase, ABC):
-    def __init__(self, reduction_shapes: Set[ReductionShape] = None, num_samples: int = None,
+    def _register_input(self, x: torch.Tensor):
+        with no_nncf_trace():
+            self._register_input_common(PTNNCFTensor(x))
```

Comment:
Can we unify this (in `PTMinMaxStatisticCollector`, `PTMixedMinMaxStatisticCollector`, `PTMeanMinMaxStatisticCollector`) ?